rpc/types_block_test.go

```diff
@@ -130,40 +132,431 @@ func TestBlock_Unmarshal(t *testing.T) {
+		require := require.New(t)
 
-			block, err := provider.BlockWithReceipts(ctx, test.BlockID)
-			require.Nil(t, err)
-			blockCasted := block.(*BlockWithReceipts)
-			require.Equal(t, test.ExpectedBlockWithReceipts, *blockCasted)
+		result, err := testConfig.provider.BlockWithReceipts(context.Background(), test.BlockID)
+		require.NoError(err, "Error in BlockWithReceipts")
+		switch resultType := result.(type) {
+		case *BlockWithReceipts:
+			block, ok := result.(*BlockWithReceipts)
+			require.True(ok, fmt.Sprintf("should return *BlockWithReceipts, instead: %T\n", result))
+			require.True(strings.HasPrefix(block.BlockHash.String(), "0x"), "Block Hash should start with \"0x\", instead: %s", block.BlockHash)
+			require.NotEmpty(block.Transactions, "the number of transactions should not be 0")
 
-		})
+			if test.ExpectedBlockWithReceipts != nil {
+				require.Equal(block.BlockHeader.BlockHash, test.ExpectedBlockWithReceipts.BlockHeader.BlockHash, "Error in BlockTxHash BlockHash")
+				require.Equal(block.BlockHeader.ParentHash, test.ExpectedBlockWithReceipts.BlockHeader.ParentHash, "Error in BlockTxHash ParentHash")
+				require.Equal(block.BlockHeader.Timestamp, test.ExpectedBlockWithReceipts.BlockHeader.Timestamp, "Error in BlockTxHash Timestamp")
+				require.Equal(block.BlockHeader.SequencerAddress, test.ExpectedBlockWithReceipts.BlockHeader.SequencerAddress, "Error in BlockTxHash SequencerAddress")
+				require.Equal(block.Status, test.ExpectedBlockWithReceipts.Status, "Error in BlockTxHash Status")
+				require.Equal(block.Transactions[len(block.Transactions)-1], test.ExpectedBlockWithReceipts.Transactions[0], "Error in BlockTxHash Transactions")
+			}
+		case *PendingBlockWithReceipts:
+			pBlock, ok := result.(*PendingBlockWithReceipts)
+			require.True(ok, fmt.Sprintf("should return *PendingBlockWithReceipts, instead: %T\n", result))
+
+			if testEnv == "mock" {
+				require.Equal(pBlock.ParentHash, test.ExpectedPendingBlockWithReceipts.ParentHash, "Error in PendingBlockWithReceipts ParentHash")
+				require.Equal(pBlock.SequencerAddress, test.ExpectedPendingBlockWithReceipts.SequencerAddress, "Error in PendingBlockWithReceipts SequencerAddress")
+				require.Equal(pBlock.Timestamp, test.ExpectedPendingBlockWithReceipts.Timestamp, "Error in PendingBlockWithReceipts Timestamp")
+				require.Equal(pBlock.Transactions, test.ExpectedPendingBlockWithReceipts.Transactions, "Error in PendingBlockWithReceipts Transactions")
+			} else {
+				require.NotEmpty(pBlock.ParentHash, "Error in PendingBlockWithReceipts ParentHash")
+				require.NotEmpty(pBlock.SequencerAddress, "Error in PendingBlockWithReceipts SequencerAddress")
+				require.NotEmpty(pBlock.Timestamp, "Error in PendingBlockWithReceipts Timestamp")
+				require.NotEmpty(pBlock.Transactions, "Error in PendingBlockWithReceipts Transactions")
+			}
+
+		default:
+			t.Fatalf("unexpected block type, found: %T\n", resultType)
+		}
 	}
```

Comment:
Can we do a final require.Equal(block, expectedBlock) at the end, just to make sure we don't miss anything?

Reply:
Done